napi/parser/scripts/build.ts

```diff
@@ -0,0 +1,66 @@
+// oxlint-disable no-console
+
+import { execSync } from "node:child_process";
+import { copyFileSync, existsSync, mkdirSync, readdirSync } from "node:fs";
+import { join } from "node:path";
+
+const parserDirPath = join(import.meta.dirname, ".."),
+  srcDirPath = join(parserDirPath, "src-js"),
+  distDirPath = join(parserDirPath, "dist");
+
+// Build with tsdown
+console.log("Building with tsdown...");
+execSync("pnpm tsdown", { stdio: "inherit", cwd: parserDirPath });
+
+// Copy native `.node` files from `src-js` to `dist`
+console.log("Copying .node files to dist...");
+for (const filename of readdirSync(srcDirPath)) {
+  if (!filename.endsWith(".node")) continue;
+  const srcPath = join(srcDirPath, filename);
+  copyFileSync(srcPath, join(distDirPath, filename));
+}
+
+// Copy bindings.js to dist (kept external from bundle)
+console.log("Copying bindings.js to dist...");
+copyFileSync(join(srcDirPath, "bindings.js"), join(distDirPath, "bindings.js"));
+
+// Copy type definitions (only the ones that are part of the public API)
+console.log("Copying type definitions...");
+copyFileSync(join(srcDirPath, "index.d.ts"), join(distDirPath, "index.d.ts"));
+// visitor.d.ts is imported by index.d.ts
+const visitDistDir = join(distDirPath, "generated", "visit");
+mkdirSync(visitDistDir, { recursive: true });
+copyFileSync(
+  join(srcDirPath, "generated", "visit", "visitor.d.ts"),
+  join(visitDistDir, "visitor.d.ts"),
+);
```

Comment:
I believe `path.join` deals with converting `/` to `\` on Windows automatically. I find this reads more clearly:

```suggestion
const visitDistDir = join(distDirPath, "generated/visit");
mkdirSync(visitDistDir, { recursive: true });
copyFileSync(
  join(srcDirPath, "generated/visit/visitor.d.ts"),
  join(visitDistDir, "visitor.d.ts"),
);
```